### DISCUSSION TITLE
Loco Positioning vs Vicon for Crazyflie

### DISCUSSION THREAD

**Discussion by sesem738** [bitcraze/discussions#1564] [Q&A]
Hello,
I'm working on integrating the Vicon motion capture system with a Crazyflie drone equipped with a mocap deck. I've encountered an issue while using the mocap_hl_commander.py script:

1. When send_full_pose=True, the Crazyflie crashes upon takeoff.
2. When send_full_pose=False, it works correctly.

I'm seeking advice on the best methods to debug this issue and improve results.
Additionally, my lab has a Loco Positioning System (LPS) that hasn't been installed yet. I'd like to know:

1. How does the LPS compare to the Vicon system in terms of accuracy and latency?
2. Would it be better to continue using the Vicon system, or should I switch to the LPS for potentially better performance?

Any insights or recommendations would be greatly appreciated. Thank you!

**Comment by gemenerik** [comment]
With LPS the positioning is done on-board the drone using signals from anchors, making it suitable for real-time localization in various environments, though with moderate precision (~10cm). In contrast, with motion capture (like Vicon) the positioning is done off-board with external cameras tracking the drone’s markers, offering very high precision (sub mm) but limited to controlled indoor settings. The off-board computation also introduces potential latency when transmitting data back to the drone.

What positioning system to use depends a lot on what your project is, but I do not think this crash alone is a reason to switch to LPS.

---

Onto the crash;

Did you read this comment?
```python
# When using full pose, the estimator can be sensitive to noise in the orientation data when yaw is close to +/- 90
# degrees. If this is a problem, increase orientation_std_dev a bit. The default value in the firmware is 4.5e-3.
```

Does the calibration look OK? Do the positioning estimates in whatever mocap software you are using make sense?
